jupyterlab_gcedetails/src/components/hardware_scaling_form.tsx

```diff
@@ -0,0 +1,243 @@
+      this.setState({
+        attachGpu: event.target.checked,
+        gpuType: event.target.checked
+          ? (ACCELERATOR_TYPES[1].value as string)
+          : NO_ACCELERATOR,
+        gpuCount: event.target.checked
+          ? (ACCELERATOR_COUNTS_1_2_4_8[0].value as string)
+          : '',
+      });
+    };
+
+    // TODO: Implement submit functionality
+    //eslint-disable-next-line @typescript-eslint/no-empty-function
+    const submitForm = () => {};
+
+    return (
+      <div className={STYLES.container}>
+        <div className={STYLES.formContainer}>
+          <span className={STYLES.title}>Hardware Scaling Limits</span>
+          <form>
+            <HardwareConfigurationDescription />
+            <div className={classes(css.flex1, STYLES.bottomPadding)}>
+              <SelectInput
+                label="Machine type"
+                name="baseMachineType"
+                value={baseMachineType}
+                options={machineTypes.map(elem => elem.base)}
+                onChange={onBaseMachineTypeChange}
+              />
+            </div>
+            <SynchronizedSliders
+              values={this.getMachineType(baseMachineType).configurations}
+              onMemoryChange={onMemoryChange}
+              onCpuChange={onCpuChange}
+              key={baseMachineType}
+            />
+            <div className={STYLES.checkboxContainer}>
+              <CheckboxInput
+                label="Attach GPUs"
+                className={STYLES.checkbox}
+                name="attachGpu"
+                checked={attachGpu}
+                onChange={onAttachGpuChange}
+              />
+            </div>
+            {attachGpu && (
+              <div
+                className={classes(css.scheduleBuilderRow, STYLES.topPadding)}
+              >
+                <div className={css.flex1}>
+                  <SelectInput
+                    label="GPU type"
+                    name="gpuType"
+                    value={gpuType}
+                    options={ACCELERATOR_TYPES.slice(1)}
+                    onChange={onGpuTypeChange}
+                  />
+                </div>
+                <div className={css.flex1}>
+                  <SelectInput
+                    label="Number of GPUs"
+                    name="gpuCount"
+                    value={gpuCount}
+                    options={ACCELERATOR_COUNTS_1_2_4_8}
+                    onChange={onGpuCountChange}
+                  />
+                </div>
+              </div>
+            )}
+          </form>
+        </div>
+        <ActionBar closeLabel="Cancel" onClick={onDialogClose}>
+          <SubmitButton
+            actionPending={false}
+            onClick={submitForm}
+            text="Save"
+          />
+        </ActionBar>
+      </div>
+    );
+  }
+}
+
+const DESCRIPTION = `The hardware scaling limits you configured will be the
+max capacity allowed for this notebook. You'll only pay for the hardware 
+resources you use. `;
```

Comment:
This description is actually incorrect in our current implementation. The user will be for the time the machine is on.

Reply:
How about this for the description: "The hardware scaling limits you configured will be the
max capacity allowed for this notebook. You'll only pay for the time the 
hardware resources are on."?

Reply:
Yeah I think that's reasonable and we can run the wording by UX design/writing at a later time.